lib/output.js

```diff
@@ -887,6 +904,21 @@ function OutputStream(options) {
         self._do_print(output, "throw");
     });
 
+    /* -----[ yield ]----- */
+
+    DEFPRINT(AST_Yield, function(self, output){
+        var star = self.is_star ? "*" : "";
+        output.print("yield" + star);
+        if (self.expression) {
+            output.space();
+            self.expression.print(output);
+        } else if (self.is_star === true) {
```

Comment:
Not sure if the code generator should be guarded against empty expressions. AST can come from multiple sources.